product_set/security/rule_product_set.xml

```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="product_set_comp_rule" model="ir.rule">
+        <field name="name">Product Set multi company rule</field>
+        <field name="model_id" ref="model_product_set" />
+        <field name="global" eval="True" />
+        <field
+            name="domain_force"
+        >['|',('company_id','=',False),('company_id','in',company_ids)]</field>
+    </record>
+    <record id="product_set_line_comp_rule" model="ir.rule">
+        <field name="name">Product Set Line multi company rule</field>
+        <field name="model_id" ref="model_product_set_line" />
+        <field name="global" eval="True" />
+        <field
+            name="domain_force"
+        >['|',('product_set_id.company_id','=',False),('product_set_id.company_id','in',company_ids)]</field>
```

Comment:
Should we put the field as related stored for performance reasons?

Reply:
you mean this way?
company_id = fields.Many2one('res.company', related='product_set_id.company_id')

<field name="domain_force">[('company_id', 'in', [False] + company_ids)]</field>

Reply:
Yes, but `store=True`